### PR TITLE
For a failed websocket upgrade, pass the HTTP response as part of the…

### PR DESCRIPTION
**Pull request by rhurey**
… error payload

### CODE REVIEW

File: src/uws_client.c [hunk 1]
```diff
@@ -1233,6 +1233,11 @@ static void on_underlying_io_bytes_received(void* context, const unsigned char*
                             LogError("Bad status (%d) received in WebSocket Upgrade response", status_code);
                             ws_open_result_detailed.result = WS_OPEN_ERROR_BAD_RESPONSE_STATUS;
                             ws_open_result_detailed.code = status_code;
+
+                            // Include the original http message so the client can determine if it cares to garner data.
+                            // Like for a redirect.
+                            ws_open_result_detailed.buffer = uws_client->stream_buffer;
```
Comment (lisaweixu) [inline src/uws_client.c:1239]: I am not sure about the life cycle of stream_buffer. I assume that you know it is still valid when the error reaches transport.cpp.  Carbon right now uses ThreadService to sync all calles into azure-c-shared-lib so we don't have to worry about thread safe issue here.

**Review by lisaweixu — APPROVED** [review]

File: src/uws_client.c (hunk 1 above)
Comment (ralph-msft) [reply src/uws_client.c:1239]: Would it also be possible to include the headers? Often times this contains important information for debugging on the server side (e.g. request IDs(

File: src/uws_client.c [hunk 2]
```diff
@@ -1097,7 +1097,7 @@ static int process_frame_fragment(UWS_CLIENT_INSTANCE *uws_client, size_t length
 
 static void on_underlying_io_bytes_received(void* context, const unsigned char* buffer, size_t size)
 {
-    WS_OPEN_RESULT_DETAILED ws_open_result_detailed = { WS_OPEN_OK, 0 };
+    WS_OPEN_RESULT_DETAILED ws_open_result_detailed = { WS_OPEN_OK, 0, NULL };
```
Comment (ralph-msft) [inline src/uws_client.c]: You should probably also set the buffSize to 0 here as well

**Review by ralph-msft — APPROVED** [review]

File: src/uws_client.c (hunk 1 above)
Comment (rhurey) [reply src/uws_client.c:1239]: It's the raw stream from the service, so it has the headers.

(The use case in the Speech SDK is to read the location header.)

**Pull request merged**